### DISCUSSION TITLE
how to import package

### DISCUSSION THREAD

**Discussion by the-pesar** [nuxt/nuxt#17883] [Questions] [answered]
how can i import `bcrypt` or other `packages` in my project ?

when i use `modules` in `nuxt.config.ts` get this error on terminal:
```
 ERROR  Cannot start nuxt:  Nuxt module should be a function: [object Object]                                                                               04:42:47

  at normalizeModule (node_modules/@nuxt/kit/dist/index.mjs:373:11)
  at installModule (node_modules/@nuxt/kit/dist/index.mjs:353:47)
  at initNuxt (node_modules/nuxt3/dist/chunks/index.mjs:1131:13)
  at async load (node_modules/nuxi/dist/chunks/dev.mjs:6732:9)
  at async Object.invoke (node_modules/nuxi/dist/chunks/dev.mjs:6771:5)
  at async _main (node_modules/nuxi/dist/chunks/index.mjs:384:7)
```

**Comment by danielroe** [comment]
What is your code?

**Reply by mihail727** [reply]
> What is your code?

`nuxt.config.ts`
```ts
export default defineNuxtConfig({
    modules: ['bcrypt'];
})
```
`@/composables/useAuth.ts`
```ts
import bcrypt from 'bcrypt';
export const hashPass = async (pass: string) => {
    return bcrypt.hash(pass, 10);
}
```

**Reply by danielroe** [reply, marked as answer]
That's not what `modules` is for. See https://v3.nuxtjs.org/docs/directory-structure/nuxt.config#modules. Just remove that line from your nuxt config and you should be good to go.

**Reply by mihail727** [reply]
> That's not what `modules` is for. See https://v3.nuxtjs.org/docs/directory-structure/nuxt.config#modules. Just remove that line from your nuxt config and you should be good to go.

ok but it doesn't work

**Reply by mihail727** [reply]
> That's not what `modules` is for. See https://v3.nuxtjs.org/docs/directory-structure/nuxt.config#modules. Just remove that line from your nuxt config and you should be good to go.

I am getting several errors:
- Could not resolve "mock-aws-s3"
- Could not resolve "aws-sdk"
- Could not resolve "nock"
- [vite] error while updating dependencies:

**Reply by mihail727** [reply]
.... Errors get out if you use bcrypt on the client side
on the server side all fine

**Reply by danielroe** [reply]
That is because `bcrypt` is a library for Node and not for the browser.

**Reply by the-pesar** [reply]
> That is because `bcrypt` is a library for Node and not for the browser.

yes its true, i use `bcryptjs` for browser
i want to import just in a page, not in The whole project
this its possible?

**Reply by mihail727** [reply]
> 

Maybe [it](https://security.stackexchange.com/questions/93395/how-to-do-client-side-hashing-of-password-using-bcrypt) will be useful for you

**Reply by mihail727** [reply]
> 

In general, the first thing that came to mind was that you can write a server-side handler that would return a hash to you at the request of the client
